### DISCUSSION TITLE
100% packet loss connecting to github.com from Tokyo, Japan over IPoE connection

### DISCUSSION THREAD

**Discussion by shirakaba** [community/community#45103] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

# Connecting from my house in Tokyo

My internet connection uses [IPoE](https://en.wikipedia.org/wiki/IPoE) and apparently may automatically decide whether to use IPv4 or IPv6 (exact text below):

> おてがる光のIPv6オプションはIPoEで接続しながら、状況を自動的判断しIPv4接続に切り替えます。 そのため、速度を活かしつつ、IPv4、IPv6を意識することなく快適にインターネットを利用できます。

I have no problem with other websites, but GitHub causes me frequent problems at all times of day; I can barely click two links in a row on GitHub without experiencing a timeout.

```sh
$ ping github.com
PING github.com (20.27.177.113): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
Request timeout for icmp_seq 2
Request timeout for icmp_seq 3
Request timeout for icmp_seq 4
Request timeout for icmp_seq 5
Request timeout for icmp_seq 6
Request timeout for icmp_seq 7
Request timeout for icmp_seq 8
Request timeout for icmp_seq 9
Request timeout for icmp_seq 10
Request timeout for icmp_seq 11
Request timeout for icmp_seq 12
Request timeout for icmp_seq 13
Request timeout for icmp_seq 14
Request timeout for icmp_seq 15
^C
--- github.com ping statistics ---
17 packets transmitted, 0 packets received, 100.0% packet loss
```

# Connecting via a VPN based in Japan

The VPN is [Private Internet Access](https://www.privateinternetaccess.com)'s "JP Streaming Optimized" VPN - it doesn't say which city it's based in. But again, it has 100% packet loss:

```sh
$ ping github.com
PING github.com (20.27.177.113): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
Request timeout for icmp_seq 2
Request timeout for icmp_seq 3
Request timeout for icmp_seq 4
Request timeout for icmp_seq 5
Request timeout for icmp_seq 6
Request timeout for icmp_seq 7
^C
--- github.com ping statistics ---
9 packets transmitted, 0 packets received, 100.0% packet loss
```

# Connecting via a VPN based in the UK

The VPN is Private Internet Access's "UK London" VPN. This time it has much less packet loss:

```sh
$ ping github.com
PING github.com (140.82.121.3): 56 data bytes
64 bytes from 140.82.121.3: icmp_seq=0 ttl=53 time=266.271 ms
64 bytes from 140.82.121.3: icmp_seq=1 ttl=53 time=267.098 ms
64 bytes from 140.82.121.3: icmp_seq=2 ttl=53 time=319.052 ms
64 bytes from 140.82.121.3: icmp_seq=3 ttl=53 time=266.902 ms
64 bytes from 140.82.121.3: icmp_seq=4 ttl=53 time=267.598 ms
64 bytes from 140.82.121.3: icmp_seq=5 ttl=53 time=266.682 ms
64 bytes from 140.82.121.3: icmp_seq=6 ttl=53 time=267.284 ms
64 bytes from 140.82.121.3: icmp_seq=7 ttl=53 time=266.754 ms
64 bytes from 140.82.121.3: icmp_seq=8 ttl=53 time=292.216 ms
64 bytes from 140.82.121.3: icmp_seq=9 ttl=53 time=266.092 ms
Request timeout for icmp_seq 10
64 bytes from 140.82.121.3: icmp_seq=11 ttl=51 time=265.139 ms
64 bytes from 140.82.121.3: icmp_seq=12 ttl=51 time=265.133 ms
64 bytes from 140.82.121.3: icmp_seq=13 ttl=51 time=265.335 ms
64 bytes from 140.82.121.3: icmp_seq=14 ttl=51 time=265.969 ms
64 bytes from 140.82.121.3: icmp_seq=15 ttl=51 time=282.268 ms
64 bytes from 140.82.121.3: icmp_seq=16 ttl=51 time=265.434 ms
^C
--- github.com ping statistics ---
18 packets transmitted, 16 packets received, 11.1% packet loss
round-trip min/avg/max/stddev = 265.133/272.202/319.052/14.055 ms
```

How can I get a more reliable connection to GitHub? Should I force it to use an IPv4 connection (and is that something I can do using my [Buffalo WSR-32000AX4S-WH](https://www.buffalo.jp/product/detail/wsr-3200ax4s-wh.html) WiFi router)?

**Comment by jge162** [comment]
@shirakaba Using IPv4 will help!!! Check if your device is currently using IPv6, you can go to a website such as "test-ipv6.com" and see what IP address is displayed. If it starts with "2604:", then your device is using IPv6.

To force your device to use IPv4, you will need to disable IPv6 on your router. The process for doing this will vary depending on the specific router you are using. Looks like your router can disable IPv4. but, check device manual to see instructions...or google process too.

**Reply by shirakaba** [reply]
> Using IPv4 will help!!!

Sounds encouraging - what makes you sure of that?

> If it starts with "2604:", then your device is using IPv6.

It's given me two addresses, so I'm not sure what to make of it:

* Your IPv4 address on the public Internet appears to be `106.*.*.*`
* Your IPv6 address on the public Internet appears to be `240b:*:*:*:*:*:*:*`

Before I start digging through my router settings in Japanese, is it fundamentally possible to disable IPv6 for a specific domain, or are my only options all-IPv4 or all-IPv6?
